tests/regression/77-mem-oob/10-oob-two-loops.c

```diff
@@ -0,0 +1,20 @@
+// PARAM: --set ana.activated[+] memOutOfBounds --enable ana.int.interval
+int main() {
+	int *p = malloc(1048 * sizeof(int));
+
+    for (int i = 0; i < 1048; ++i) {
+		p[i] = __VERIFIER_nondet_int(); //NOWARN
+	}
+
+	int *q = p;
+
+	while (*q >= 0 && q < p + 1048 * sizeof(int)) { //WARN
+		if (__VERIFIER_nondet_int()) {
+			q++;
+		} else {
+			(*q)--; //WARN
```

Comment:
Can you explain why this is an invalid deref?

Reply:
My assumption here is that the condition `q < p + 1048 * sizeof(int)` could lead to an overflow, but I might be wrong. My line of thought is that the pointer arithmetic with `p` should be only `p + 1048` in order to be correct.

Reply:
Ah, I see!

Reply:
I think that the `q++` part could keep incrementing beyond the bounds of `q`'s contents, which could lead to an invalid-deref in `(*q)--`